components/group-mgt/org.wso2.carbon.device.mgt.group.core/src/main/java/org/wso2/carbon/device/mgt/group/core/dao/GroupDAOImpl.java

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright (c) 2014, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.device.mgt.group.core.dao;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.device.mgt.group.common.DeviceGroup;
+import org.wso2.carbon.device.mgt.group.core.internal.DeviceGroupBroker;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+public class GroupDAOImpl implements GroupDAO {
+
+    private static final Log log = LogFactory.getLog(GroupDAOImpl.class);
+
+    @Override public int addGroup(DeviceGroup deviceGroup, int tenantId) throws GroupManagementDAOException {
+        PreparedStatement stmt = null;
+        ResultSet rs;
+        int groupId = -1;
+        try {
+            Connection conn = GroupManagementDAOFactory.getConnection();
+            String sql = "INSERT INTO DM_GROUP(DESCRIPTION, NAME, DATE_OF_ENROLLMENT, DATE_OF_LAST_UPDATE, "
+                    + "OWNER, TENANT_ID) VALUES (?, ?, ?, ?, ?, ?)";
+            stmt = conn.prepareStatement(sql);
+            stmt.setString(1, deviceGroup.getDescription());
+            stmt.setString(2, deviceGroup.getName());
+            stmt.setLong(3, new Date().getTime());
+            stmt.setLong(4, new Date().getTime());
+            stmt.setString(5, deviceGroup.getOwner());
+            stmt.setInt(6, tenantId);
+            stmt.executeUpdate();
+            rs = stmt.getGeneratedKeys();
+            if (rs.next()) {
+                groupId = rs.getInt(1);
+            }
+            return groupId;
+        } catch (SQLException e) {
+            String msg = "Error occurred while adding deviceGroup " +
```

Comment:
Fixed in https://github.com/wso2/carbon-device-mgt/commit/e1cd4b1907f419f974545fc214c58cd9f247f2b2